src/lib.rs

```diff
@@ -27,15 +27,18 @@
 #![deny(unused_mut)]
 #![deny(missing_docs)]
 
+#![cfg_attr(all(not(test), not(feature = "std")), no_std)]
```

Comment:
We should really add an "integration test" that is not a test and doesn't use `std` to see if it actually compiles. I'm not sure what's the best way to do this. Maybe we don't need to treat tests in a special way since `std` is active by default (so tests will never run without `std` anyway in practice)?

Reply:
Oh, yeah, we should add `--no-default-features` to travis.

Compiling with `cargo test` compiles twice, once with `#[cfg(test)]` and once without. (That's why errors are doubled up usually.)